dbms/src/Common/tests/gtest_sensitive_data_masker.cpp

```diff
@@ -0,0 +1,220 @@
+        EXPECT_EQ(
+            std::string(e.what()),
+            "SensitiveDataMasker: cannot compile re2: **, error: no argument for repetition operator: *. Look at "
+            "https://github.com/google/re2/wiki/Syntax for reference.");
+        EXPECT_EQ(e.code(), DB::ErrorCodes::CANNOT_COMPILE_REGEXP);
+    }
+    /* catch (...) { // not needed, gtest will react unhandled exception
+        FAIL() << "ERROR: Unexpected exception thrown: " << std::current_exception << std::endl; // std::current_exception is part of C++11x
+    } */
+
+    EXPECT_EQ(maskerbad.rulesCount(), 0);
+    EXPECT_EQ(maskerbad.wipeSensitiveData(x), 0);
+
+    std::istringstream xml_isteam(R"END(<?xml version="1.0"?>
+<clickhouse>
+    <query_masking_rules>
+        <rule>
+            <name>hide SSN</name><!-- by default: it will use xml path, like query_masking_rules.rule[1] -->
+            <regexp>[0-9]{3}-[0-9]{2}-[0-9]{4}</regexp><!-- mandatory -->
+            <replace>000-00-0000</replace><!-- by default - six asterisks (******) -->
+        </rule>
+        <rule>
+            <name>hide root password</name>
+            <regexp>qwerty123</regexp>
+        </rule>
+        <rule>
+            <regexp>(?i)Ivan</regexp>
+            <replace>John</replace>
+        </rule>
+        <rule>
+            <regexp>(?i)Petrov</regexp>
+            <replace>Doe</replace>
+        </rule>
+        <rule>
+            <name>hide email</name>
+            <regexp>(?i)[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}</regexp>
+            <replace>hidden@hidden.test</replace>
+        </rule>
+        <rule>
+            <name>remove selects to bad_words table</name>
+            <regexp>^.*bad_words.*$</regexp>
+            <replace>[QUERY IS CENSORED]</replace>
+        </rule>
+    </query_masking_rules>
+</clickhouse>)END" );
+
+    Poco::Util::XMLConfiguration * xml_config(new Poco::Util::XMLConfiguration(xml_isteam));
+    SensitiveDataMasker masker_xml_based( *xml_config, "query_masking_rules" );
+    std::string top_secret = "The e-mail of IVAN PETROV is kotik1902@sdsdf.test, and the password is qwerty123";
+    EXPECT_EQ(masker_xml_based.wipeSensitiveData(top_secret), 4);
+    EXPECT_EQ(top_secret, "The e-mail of John Doe is hidden@hidden.test, and the password is ******");
+
+    top_secret = "SELECT * FROM bad_words";
+    EXPECT_EQ(masker_xml_based.wipeSensitiveData(top_secret), 1);
+    EXPECT_EQ(top_secret, "[QUERY IS CENSORED]");
+
+#ifndef NDEBUG
+    masker_xml_based.printStats();
+#endif
+
+    try
+    {
+        std::istringstream xml_isteam_bad(R"END(<?xml version="1.0"?>
+<clickhouse>
+    <query_masking_rules>
+        <rule>
+            <name>test</name>
+            <regexp>abc</regexp>
+        </rule>
+        <rule>
+            <name>test</name>
+            <regexp>abc</regexp>
+        </rule>
+    </query_masking_rules>
+</clickhouse>)END" );
+
+        Poco::Util::XMLConfiguration * xml_config1(new Poco::Util::XMLConfiguration(xml_isteam_bad));
```

Comment:
PVS complaing about releasing that memory

Reply:
Yes, because object should be owned by someone (Poco::AutoPtr in this case).